packages/connect/src/SmoldotProvider/SmoldotProvider.ts

```diff
@@ -280,34 +281,33 @@ export class SmoldotProvider implements ProviderInterface {
           }
         }
       })
-      .then((client: smoldot.SmoldotClient) => {
-        this.#client = client;
-        this.#connectionStatePingerId = setInterval(
-          this.#checkClientPeercount, this.healthPingerInterval);
-      })
-      .catch((error: Error) => {
-        this.emit('error', error);
-        return Promise.reject(error);
-      });
+      this.#connectionStatePingerId = setInterval(
+      this.#checkClientPeercount, this.healthPingerInterval);
+    } catch(error: unknown) {
+      this.emit('error', error);
+    }
   }
 
   /**
    * @description Manually "disconnect" - drops the reference to the WASM client
    */
   // eslint-disable-next-line @typescript-eslint/require-await
   public async disconnect(): Promise<void> {
-    if (this.#client) {
-      this.#client = undefined;
-    }
-
-    if (this.#connectionStatePingerId !== null) {
-      clearInterval(this.#connectionStatePingerId);
-    }
+    try {
+      if (this.#client) {
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        this.#client.terminate;
+      }
 
-    this.#isConnected = false;
-    this.emit('disconnected');
+      if (this.#connectionStatePingerId !== null) {
+        clearInterval(this.#connectionStatePingerId);
+      }
 
-    return Promise.resolve();
+      this.#isConnected = false;
+      this.emit('disconnected');
```

Comment:
If you are using a `try`/`catch` these lines ought to be in a `finally` block ... we want to clear the interval / set the isconnected state and emit disconnected regardles of whether there is an error or not.  This provider is no longer usable without trying to reconnect